gnatsd/tests/test.sh

```diff
@@ -1,34 +1,32 @@
 #!/bin/sh
+#/ Usage: test.sh <pkg_ident>
+#/
+#/ Example: test.sh core/php/7.2.8/20181108151533
+#/
 
-TESTDIR="$(dirname "${0}")"
-PLANDIR="$(dirname "${TESTDIR}")"
-SKIPBUILD=${SKIPBUILD:-0}
+set -euo pipefail
 
-hab pkg install core/bats --binlink
+source bin/ci/test_helpers.sh
+
+if [[ -z "${1:-}" ]]; then
+  grep '^#/' < "${0}" | cut -c4-
+	exit 1
+fi
 
+TEST_PKG_IDENT="${1}"
+export TEST_PKG_IDENT
+hab pkg install core/bats --binlink
 hab pkg install core/busybox-static
 hab pkg binlink core/busybox-static ps
 hab pkg binlink core/busybox-static netstat
 hab pkg binlink core/busybox-static wc
 hab pkg binlink core/busybox-static uniq
+hab pkg install "${TEST_PKG_IDENT}"
 
-source "${PLANDIR}/plan.sh"
-
-if [ "${SKIPBUILD}" -eq 0 ]; then
-  # Unload the service if its already loaded.
-  hab svc unload "${HAB_ORIGIN}/${pkg_name}"
 
-  set -e
-  pushd "${PLANDIR}" > /dev/null
-  build
-  source results/last_build.env
-  hab pkg install "results/${pkg_artifact}" --binlink --force
-  hab svc load "${pkg_ident}"
-  popd > /dev/null
-  set +e
+ci_ensure_supervisor_running
+ci_ensure_service_loaded "$TEST_PKG_IDENT"
 
-  # Give some time for the service to start up
-  sleep 10
-fi
+bats "$(dirname "${0}")/test.bats"
 
-bats "${TESTDIR}/test.bats"
+hab svc unload "${TEST_PKG_IDENT}"
```

Comment:
We could maybe have a helper for this - and one point that's been discussed on previous PRs is whether we should hide a failure here. 
If the tests all pass, we might want the build step to pass. Or we might want the svc unload to be an implicit test for this package.

Reply:
I almost did this, but decided to skip it for now. In the case the tests fail, it will never get called. If it does get called, the container it runs in in CI might go away before it's fully unloaded because it's async. 

In the context of local testing, it will get unloaded (assuming tests pass), and I'm not entirely sure that we need to wait for it to unload in order for CI to finish since we should be running in single-use containers, so the function would essentially be a wrapper around `hab svc unload` with no added value.

Not to say there isn't something we could do to improve this, but short term I think `hab svc unload` is sufficient.

Reply:
Ah, thanks for explaining!